src/js/classes/Streamer.js

```diff
@@ -18,12 +18,34 @@ import '../typedefs';
 import selectRepresentations from '../utils/selectRepresentations';
 import getRepresentationMimeString from '../utils/getRepresentationMimeString';
 
+/**
+ * What maximum percentage of total frames are OK to be dropped for us to still
+ * consider the playback experience acceptable?
+ */
+const HEALTHY_PERCENTAGE_OF_DROPPED_FRAMES = 0;
```

Comment:
Do your local tests look good with `0`?

Reply:
@beaufortfrancois Yes, but mostly just because of the algorithm used to determine if the playback is smooth.

We compare this number to a minimum value from the several last dropped frames readings. The sample size is determined by `DROPPED_FRAMES_PERCENTAGE_SAMPLE_SIZE`.

Essentially now we're taking 3 samples one second apart and if all of them report non-zero amount of dropped frames, we load a less demanding source. If some samples report no dropped frames, we assume the playback is nearly smooth and allow it to continue.